src/main/java/com/matera/account/account/AccountController.java

```diff
@@ -0,0 +1,52 @@
+package com.matera.account.account;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+public class AccountController {
+
+    @Autowired
+    private AccountService accountService;
+
+    @GetMapping
+    // List all Accounts
+    public List<Account> getAll() {
+        return accountService.findAll();
+    }
+
+    @GetMapping("/clients/{clientId}/accounts")
+    // List all Accounts for a specific client
+    public List<Account> getAccountsByClientId(@PathVariable("clientId") UUID clientId) {
+        return accountService.findClientAccountsByClientId(clientId);
+    }
+
+    @GetMapping("/clients/{clientId}/accounts/{accountId}")
+    // Find Account by accountId
+    public Account getAccountByAccountId(@PathVariable("clientId") UUID clientId, @PathVariable("accountId") UUID accountId) {
+        return accountService.findAccountByAccountId(clientId, accountId); 
+    }
+
+    @PostMapping("/client/{clientId}/accounts")
+    // Create account for specific client
+    public Account insertAccount(@PathVariable("clientId") UUID clientId, @RequestBody @Valid AccountDTO account) {
```

Comment:
For validation here, you need to annotate the DTO class attributes, here are some documentation on it, https://www.baeldung.com/javax-validation

but this is not needed at this stage :) so we are good here